programs/stake/src/stake_state.rs

```diff
@@ -3689,4 +3692,86 @@ mod tests {
         // Test another staking action
         assert_eq!(stake_keyed_account.deactivate(&clock, &new_signers), Ok(()));
     }
+
+    #[test]
+    fn test_redelegate_considers_withdraws() {
```

Comment:
nice name.  covering an increase in account balance might reassure.

Reply:
Good point!  I'll add that, and make the name even better :wink: